src/components/ChatBox.tsx

```diff
@@ -0,0 +1,281 @@
+import React, { useEffect, useRef, useState } from "react";
+import ChatTopBar from "./ChatTopBar";
+import ChatMessages from "./ChatMessages";
+import ChatBottom from "./ChatBottom";
+import "../scss/components/_chatBox.scss";
+import { Client, Stomp } from "@stomp/stompjs";
+import sockJS from "sockjs-client";
+import store from "../store/RootStore";
+import { api } from "../api/api";
+import { ChatMessage, WebsocketResponse } from "../types/chat.types";
+
+interface ChatBoxProperties {
+  recipientEmail: string;
+  profilePicture: string | null;
+  onClose: () => void;
+  onMinimize: () => void;
+  name: string;
+  surname: string;
+  profile: string;
+}
+
+const ChatBox: React.FC<ChatBoxProperties> = ({
+  recipientEmail,
+  profilePicture,
+  onClose,
+  onMinimize,
+  name,
+  surname,
+  profile,
+}) => {
+  const [wsClient, setWsClient] = useState<Client | null>(null);
+  const [doesChatRoomExist, setDoesChatRoomExist] = useState<boolean | null>(
+    null
+  );
+  const wsClientRef = useRef(wsClient);
+  const [chatId, setChatId] = useState<number | null>(null);
+  const [messages, setMessages] = useState<ChatMessage[]>([]);
+  const messagesRef = useRef(messages);
+  const [lastSeenMessage, setLastSeenMessage] = useState<number | undefined>();
+  const { timeZone } = Intl.DateTimeFormat().resolvedOptions();
+
+  useEffect(() => {
+    wsClientRef.current = wsClient;
+  }, [wsClient]);
+  useEffect(() => {
+    initWebsocketConnection();
+    return () => {
+      disconnectWebSocket();
+    };
+  }, [recipientEmail]); //to change, when chat with other recipient selected
+
+  const initWebsocketConnection = () => {
+    const socket = new sockJS(
+      `http://localhost:8081/ws?token=${store.user.jwtToken}`
+    );
+    const client = Stomp.over(() => socket);
+    const onConnect = () => {
+      setWsClient(client);
+      console.log("WS client connected");
+    };
+    client.connect({}, onConnect);
+  };
+
+  useEffect(() => {
+    if (
+      wsClient !== null &&
+      store.user.profile !== null &&
+      store.user.profile?.selected_profile !== null
+    ) {
+      checkIfChatRoomExists();
+    }
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [wsClient, store.user.profile?.selected_profile]);
+
+  const checkIfChatRoomExists = async () => {
+    try {
+      const data = await api.getChatRoomWithGivenUser(recipientEmail, timeZone);
+      setChatId(data.id);
+      setDoesChatRoomExist(true);
+    } catch (error: unknown) {
+      if (error instanceof Error) {
+        const regex = /Status code: ([0-9]{3})/;
+        const match = error.message.match(regex);
+        if (match !== null && match[1] === "404") {
+          setDoesChatRoomExist(false);
+        }
+      }
+    }
+  };
+
+  useEffect(() => {
+    if (doesChatRoomExist === true && chatId !== null) {
```

Comment:
Tutaj też skróciłbym, na pewno `doesChatRoomExist` bo jest typu `boolean`, z `chatId` to zależy czy chcemy dopuszczać sytuacje gdy `chatId` jest równe 0/NaN, ale raczej nie więc też skróciłbym:  
```suggestion
    if (doesChatRoomExist && chatId) {
```